extensions/src/AWSSDK.Extensions.Configuration.AWSSystemsManager/AWSSystemsManagerProcessor.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Amazon.Extensions.NETCore.Setup;
+using Amazon.SimpleSystemsManagement;
+using Amazon.SimpleSystemsManagement.Model;
+
+namespace AWSSDK.Extensions.Configuration.AWSSystemsManager
+{
+    internal interface IAWSSystemsManagerProcessor
+    {
+        Task<List<Parameter>> GetParametersByPath(AWSOptions awsOptions, string path);
+    }
+
+    internal class AWSSystemsManagerProcessor : IAWSSystemsManagerProcessor
+    {
+        public async Task<List<Parameter>> GetParametersByPath(AWSOptions awsOptions, string path)
+        {
+            using (var client = awsOptions.CreateServiceClient<IAmazonSimpleSystemsManagement>())
+            {
+                var parameters = new List<Parameter>();
+                string nextToken = null;
+                do
+                {
+                    var response = await client.GetParametersByPathAsync(new GetParametersByPathRequest { Path = path, Recursive = true, WithDecryption = true, NextToken = nextToken });
```

Comment:
Actually I have some more details that we should consider
>If you specify a parameter hierarchy using forward slashes, the hierarchy can have a maximum of five levels, as shown in this example: /Level-1/Level-2/Level-3/Level-4/Level-5/parameter-name
Also, ":" isn't allowed. Just ".-_"

So if we want to allow collapsed hierarchies or allow depths greater than 5 then we will need to support a different convention. My suggestion is __ since it's consistent with `.AddEnvironmentVariables`
>For hierarchical config values specified in environment variables, a colon (:) may not work on all platforms. Double underscore (__) is supported by all platforms.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.1&tabs=basicconfiguration#configuration-by-environment

However, I'm open to other suggestions.

Reply:
Awaiting Input

Reply:
When I was suggesting this I was thinking if recusive was turned off, which would not be the default, we would only add the parameters at the root to the configuration. So in your example `IOptions<AppSettings>` and SSM with

```
/approot/appsettings/connectionstrings/connection1
/approot/appsettings/connectionstrings/connection2
/approot/appsettings/url
/approot/appsettings/port
```

Only url and port would be added to the `IOptions` object when recursive was set to false. I'm not sure what the use case would be for doing that, I just wondered if we should provide that switch to users. I'm also good with not having it and if there is an ask later we can added.

Reply:
Your case won't actually work as you're are expecting, at least not without a rework that adds some extra settings and complexity. I've tried to type up all the reasoning why a couple times, but each time I deleted it because it started getting too complex and I felt like it would cause more questions than answers. 

I think there are a few ways to approach this:
  1. Play around with it in the code and I think you'll see why
  2. We have an actual conversation somewhere and I can explain (IM or voice)
  3. Just trust me :)

Btw, if you do want to have a conversation at some point just let me know. I can reach out to our TAM to put us into contact with each other. This definitely isn't necessary, but if you would find it beneficial at any point it is an option